### ISSUE TITLE
Dalek Death?

### ISSUE THREAD

**Issue by dannykopping** [dalekjs/dalek#152]
Hey - is this project dead? Almost 3 months without a commit to either this or Skaro. Should we be looking for alternatives? We use this at work and just need some indication either way. Thanks guys :+1:

**Comment by elliotboney** [comment]
I too am wondering, been building out tons of tests for work and starting to get a little nervous. Thanks!

**Comment by Munter** [comment]
If you are invested in an eco system and want to see if move forward the best strategy is to help keep it alive. I bet pull requests from active contributors would be more than welcome.

**Comment by dannykopping** [comment]
But to which project? Dalek or Skaro? It'd just be good to get a sense of where they're at.

**Comment by elliotboney** [comment]
Yeah I'd love to contribute too if I can get pointed to the right place.

**Comment by jackkettle** [comment]
+1

**Comment by danielbayerlein** [comment]
Can someone make a statement? /cc @anselmh @asciidisco @ryanzec @DalekJS

**Comment by asciidisco** [comment]
@rodneyrehm summed it up here: https://github.com/dalekjs/skaro/issues/1#issuecomment-78226497

**Comment by danielbayerlein** [comment]
@asciidisco Thanks! :+1:

**Comment by dannykopping** [comment]
@asciidisco great! thanks for the update

**Comment by asciidisco** [comment]
I´m working on a more detailed blog post, that explains what happened in the past, what´s going on today & what will be the future of project. Also I elaborate a lot on the why progress slowed down a lot.
I´ll keep you posted.

**Comment by rafinskipg** [comment]
Just for saying. I have been playing around with dalek. I think its a great project, it does what its needed and no more.

Other projects lile theintern are too big, or casperjs has an awfull syntax. But dalek is just fine.

Dont let it die xD

**Comment by rodneyrehm** [comment]
> Other projects lile theintern are too big, or casperjs has an awfull syntax

* what makes [Intern](https://theintern.github.io/#compare) look big? (besides out of the box support for pretty much everything)
* what's so awful about [CasperJS](http://docs.casperjs.org/en/latest/testing.html#testing)?

This actually is valuable feedback. Share your thoughts!

> But dalek is just fine.

Actually it isn't, at least not under the hood. There are so many problems with dalek - in respect to evolving the base, not necessarily to using it - that we decided to start from scratch. We intend to make Dalek much more powerful, while trying to keep it as simple as possible. We'll see how that turns out.

Do you mind telling us *exactly* what makes you like DalekJS better than Intern and CasperJS?

> Dont let it die xD

we don't intend to, but we do intend to give it a fundamental overhaul.

**Comment by rafinskipg** [comment]
HI @rodneyrehm 

- The intern looks big because it has a lot of features. As you say, it supports nearly everything, so when i was looking for a "functional testing" tool, having already a unit testing tool configured, like karma, i just need something concrete.
But probably is the tool that we have to use right now or until the new version of dalekjs is complete.

- CasperJS it's not mainly focused to testing, or it doesn't seem so. It seems just a scripting tool. DalekJS tests are pretty good in comparision. Also CasperJS doesn't seem to support any other navigator than phantomjs

What did I liked from DalekJS? 
- The drawing xD
- That it has a cli, and a gulp plugin
- It is simple. I just setted my first tests with a super easy way to do screenshots in a few minutes.
- It is for tests only.

What i think it needs:
- Tests for itself!
- Some mobile capabilities

**Comment by rafinskipg** [comment]
@rodneyrehm there's any other alternative to dalekjs which does not run with selenium?

**Comment by Eternal-Infinity** [comment]
Are there some news to this topic?
Either an update to "what's going on with DalekJS" or at least the "are there worthy alternatives to DalekJS?"?

I too have to say DalekJS is a great tool and it was exactly what I was looking for.
It would be sad to just let it die.

**Comment by ryanzec** [comment]
@rafinskipg You can use https://github.com/angular/protractor with chromedriver and not use selenium (though it has that option too).  While https://github.com/angular/protractor was originally designed for AngularJS, it doesn't take a lot of work to get it to work with any codebase.

I have switched over to Protractor as my browser runner for testing and though I do like DalekJS (heck, I have a number of commits to the project), it's lack of core development has left me to worried to really invest that much into it.  It's the downside of open source projects that don't have company backing can fall into, they can fall to the side for a while as real life takes priority.

**Comment by stehuebn** [comment]
Just made my first steps with dalek. And I like it very much! Got me just few minutes to make a check-login-success-and-checkout-again-test (worked fine for me with phantomjs).
I really hope the project stays alive - as dalek or skaro - the name is not important! Just keep it simple! :-)

**Comment by rodneyrehm** [comment]
People looking for alternatives to Dalek/Skaro may give [The Intern](https://theintern.github.io/) a shot. I've used it in a library project recently - and have to say it already provides most of what we were trying to do with Dalek in the first place. Of course it has room for improvemenet, but it's off to a pretty good start. Especially the combined handling of unit- and functional-tests makes it quite nice.

**Comment by rodrigomaia17** [comment]
Hey guys, please dont let dalek die, this project is awesome!

(And I can't tolerate selenium in js projects anymore =/ )

**Comment by jmcoimbra** [comment]
Screenshot is easy in dalek, just saying.

ref.: https://github.com/theintern/intern/issues/368

**Comment by ryanzec** [comment]
Well unfortunately it does seem like this project is dead (or on life support), at least to me. It has been over 1 year since anything was done to the https://github.com/dalekjs/skaro repo which was supposed to be the new API for Dalek. I was really hopeful for this project when it first came out but now there are a number of other choices that people can choose from that seem pretty good.  You have http://www.protractortest.org/#/ or https://theintern.github.io/ that seem pretty popular. I myself have moved towards using http://webdriver.io/.

This projects looked really cool back when it was released and there was not a lot of other projects that were mature at the time.  Now this project has really fallen behind as far as activity and even though it is pretty functionality, I just can't use it as it does not seem like it is going to survive very long (if you still consider it surviving).

**Comment by rodrigomaia17** [comment]
hey @ryanzec  are you satisfied with webdriver (and Selenium)? I feel horrible for having to install java only to test a javascript app.

**Comment by ryanzec** [comment]
I was pretty happy with the functionality of it when I used it a while back (have not been writing as many tests like that as of late but it was able to do everything I needed when I was using it).

**Comment by techtonik** [comment]
Are there any libs that are close to Dalek test syntax? Dalek is no more compatible with latest Grunt, so I need to rewrite its tests.
